core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -685,9 +794,30 @@ public boolean supportIsDescendant() {
         @Override
         @Restricted(NoExternalUse.class)
         public boolean isDescendant(String potentialChildRelativePath) throws IOException {
+            if (potentialChildRelativePath.equals("") && cacheDescendant) {
+                return true;
+            }
+
             if (new File(potentialChildRelativePath).isAbsolute()) {
                 throw new IllegalArgumentException("Only a relative path is supported, the given path is absolute: " + potentialChildRelativePath);
             }
+
+            // shortcut for direct child to avoid the complexity of the whole computation
+            // as we know that a file that is a direct descendant of its parent can only be descendant of the root
+            // if the parent is descendant
+            FilePath directChild = this.f.child(potentialChildRelativePath);
+            if (Objects.equals(directChild.getParent(), this.f)) {
+                try {
+                    boolean isDirectDescendant = this.f.isDescendant(potentialChildRelativePath);
+                    if (isDirectDescendant) {
+                        return true;
+                    }
+                    // not a return false because you can be a non-descendant of your parent but still
+                    // inside the root directory
+                }
+                catch (InterruptedException e) {
```

Comment:
I added a `return` as we do not want to continue actually.

Reply:
It will suppress the interruption logic for the thread. I do not think it is good to suppress that, and I would rather prefer to see it propagates as far as API compatibility requirements allow

Reply:
My comment is still actual here

Reply:
(I saw the conversation marked as resolved but not sure who did that)
It's considered in the next commit, wrapping it inside an `IOException`.